services/emailService.js

```diff
@@ -0,0 +1,58 @@
+const nodemailer = require("nodemailer");
+const htmlToText = require("nodemailer-html-to-text").htmlToText;
+const path = require("path");
+
+let {
+	EMAIL_USER,
+	EMAIL_PASS,
+	EMAIL_HOST,
+	EMAIL_SENDER,
+	WISH_MAIL_SUBJECT
+} = require("../config/index");
+const { logger, toTitleCase } = require("../utility/helpers");
+const { generateMailHtml } = require("../utility/emailTemplate");
+
+const transporter = nodemailer.createTransport({
+	type: "SMTP",
+	host: EMAIL_HOST,
+	secure: true,
+	debug: true,
+	port: 465,
+	auth: {
+		user: EMAIL_USER,
+		pass: EMAIL_PASS
+	}
+});
+
+module.exports.sendWishMail = async (email, data) => {
+	let imagePath = path.join(
+		__dirname,
+		`../assets/images/wish_${data._id}.png`
+	);
+	let { name } = data;
+	let mailOptions = {
+		from: `DSCKIET <${EMAIL_SENDER}`,
+		to: `${data.name} <${data.email}>`,
+		cc: "dsckiet@gmail.com",
+		subject: WISH_MAIL_SUBJECT,
+		html: generateMailHtml(toTitleCase(String(name).trim().split(" ")[0])),
+		attachments: [
+			{
+				filename: `wish_${data._id}.png`,
+				path: imagePath,
```

Comment:
What is `imagePath` here? How is the image being uploaded and accessible?

Reply:
- for every user, an image is being generated with his name and image using node canvas and being saved as wish_<userid>.png in assets/images folder

Reply:
so is there a way to not save files and use the generated image as it is in the attachment?

Reply:
will be sending only buffer